fdbclient/S3BlobStore.actor.cpp

```diff
@@ -1943,23 +1957,94 @@ ACTOR Future<int> readObject_impl(Reference<S3BlobStoreEndpoint> bstore,
                                   void* data,
                                   int length,
                                   int64_t offset) {
-	if (length <= 0)
-		return 0;
 	wait(bstore->requestRateRead->getAllowance(1));
+	// Build up a trace event for the read object request to log on exception.
+	state TraceEvent te("S3BlobStoreReadObject");
+	te.detail("Bucket", bucket)
+	    .detail("Length", length)
+	    .detail("Offset", offset)
+	    .detail("Host", bstore->host)
+	    .detail("Port", bstore->service);
 
-	state std::string resource = constructResourcePath(bstore, bucket, object);
-	state HTTP::Headers headers;
-	headers["Range"] = format("bytes=%lld-%lld", offset, offset + length - 1);
-	Reference<HTTP::IncomingResponse> r =
-	    wait(bstore->doRequest("GET", resource, headers, nullptr, 0, { 200, 206, 404 }));
-	if (r->code == 404)
-		throw file_not_found();
-	if (r->data.contentLen !=
-	    r->data.content.size()) // Double check that this wasn't a header-only response, probably unnecessary
-		throw io_error();
-	// Copy the output bytes, server could have sent more or less bytes than requested so copy at most length bytes
-	memcpy(data, r->data.content.data(), std::min<int64_t>(r->data.contentLen, length));
-	return r->data.contentLen;
```

Comment:
This old code looks like a bug?

Reply:
Zhe is referring to our return of the min of r->data.contentLen and the length param...  They look to be the same. contentLength is assigned the length down in doRequest.